src/components/auth/forgotPassword/forgotPasswordForm.component.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import {
+  View,
+  ViewProps,
+} from 'react-native';
+import {
+  ThemedComponentProps,
+  ThemeType,
+  withStyles,
+} from '@kitten/theme';
+import {
+  Button,
+  ValidationInput,
+} from '@src/components/common';
+import { EmailIconFill } from '@src/assets/icons';
+
+export interface ForgotPasswordFormType {
+  email: string;
+}
+
+interface ComponentProps {
+  onSubmit: (value: ForgotPasswordFormType) => void;
+}
+
+export type ForgotPasswordFormProps = ThemedComponentProps & ViewProps & ComponentProps;
+
+interface State {
+  email: string;
+  emailValid: boolean;
+}
+
+class ForgotPasswordFormComponent extends React.Component<ForgotPasswordFormProps, State> {
+
+  public state: State = {
+    email: '',
+    emailValid: false,
+  };
+
+  private emailValidator: RegExp = /\S+@\S+\.\S+/;
+
+  private onSubmitButtonPress = () => {
+    const { email } = this.state;
+
+    this.props.onSubmit({ email });
+  };
+
+  private onEmailValidationResult = (emailValid: boolean, email: string) => {
+    this.setState({
+      emailValid,
+      email,
+    });
+  };
+
+  private isFormValid = (): boolean => {
```

Comment:
Does it make sense to create method for checking boolean variable?

Reply:
just because in other `form` components it exists

Reply:
Do they implement the same interface?